lib/pem.ml

```diff
@@ -77,13 +58,19 @@ module Cs = struct
   let parse_with_errors data = combine (lines data)
 
   let unparse ~tag value =
-    let rec split_at_64 acc = function
-      | x when length x <= 64 -> List.rev (x :: acc)
-      | x -> let here, rest = split x 64 in
-        split_at_64 (here :: acc) rest
+    let split_at_64 data =
+      let dlen = length data in
+      let rec go acc off =
+        if dlen - off <= 64 then
+          List.rev (sub data off (dlen - off) :: acc)
+        else
+          let chunk = sub data off 64 in
+          go (chunk :: acc) (off + 64)
+      in
+      go [] 0
     in
     let raw = Base64.encode_string value in
-    let pieces = split_at_64 [] raw in
+    let pieces = split_at_64 raw in
     let nl = "\n" in
```

Comment:
what I'm not 100% sure about, in our decode we support both \n and \r\n -- while in our encode here we always use \n -- should that be different?

I don't know precisely how to do that, given that on unix we may have \r\n as well. so maybe the best is to remove the \r support?